File: node/test/api/jdy/app.test.js
/**
 * Copyright (c) 2015-2022, FineX, All Rights Reserved.
 * @author Thomas.Zhuang
 * @date 2022/05/17
 */

import { AppApiClient } from '../../../src/api/jdy/app';
import { API_KEY, HOST } from '../../../src/constants/http_constant';

export const appTest = 'appTest';

let appId;
describe('app api test', () => {
    const api = new AppApiClient(API_KEY, HOST, 'v5');

    test('appList', async () => {
        const appList = await api.appList({});
        console.log('appList result:' + JSON.stringify(appList));
        expect(appList['apps']).toBeTruthy();
        appId = appList['apps'][0].app_id;
    });

    test('entryList', async () => {
        const entryList = await api.entryList(appId, {});
        console.log('entryList result:' + JSON.stringify(entryList));
        expect(entryList.forms).toBeTruthy();
    });
});

File: csharp/test/api/jdy/App.Test.cs
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/01
*/

using Src.Api.Jdy;
using Src.Base;

namespace Test;

[TestCaseOrderer("XUnit.Project.Orderers.AlphabeticalOrderer", "Test")]
public class AppTest
{
    public static string appId = "";
    public static string entryId = "";

    private static AppApiClient api = new AppApiClient(Global.Host, Global.ApiKey);

    [Fact]
    public async void TestAppList()
    {
        var appList = await api.appList(new Dictionary<string, object>());
        Assert.True(appList != null);
        Console.WriteLine(appList);
        appId = appList.Value.GetProperty("apps")[0].GetProperty("app_id").GetString();
    }

    [Fact]
    public async void TestEntryList()
    {
        var entryList = await api.entryList(appId, new Dictionary<string, object>());
        Assert.True(entryList != null);
        entryId = entryList.Value.GetProperty("forms")[0].GetProperty("entry_id").GetString();
        Console.WriteLine(entryList);
    }
}